src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1578,6 +1584,80 @@ private static bool HasAnonymousFunctionConversion(BoundExpression source, TypeS
 
             return IsAnonymousFunctionCompatibleWithType((UnboundLambda)source, destination) == LambdaConversionResult.Success;
         }
+
+        internal enum ConstructibleCollectionTypeKind
+        {
+            None = 0,
+            Array,
+            Span,
+            ReadOnlySpan,
+            CollectionInitializer,
```

Comment:
nit: consider adding a prototype comment about expanding this to constructible types once we settle on that.

Reply:
Will leave the enum as is. We won't forget to add entries for other constructible collection type kinds.